fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/serialization/CalculateLoanScheduleQueryFromApiJsonHelper.java

```diff
@@ -124,8 +124,13 @@ public void validate(final String json) {
             repaymentsStartingFromDate = this.fromApiJsonHelper.extractLocalDateNamed(repaymentsStartingFromDateParameterName, element);
         }
 
-        validateRepaymentsStartingFromDateIsAfterDisbursementDate(dataValidationErrors, expectedDisbursementDate,
-                repaymentsStartingFromDate);
+        final String activateOnApprovalParameterName = "activateOnApproval";
+        final Boolean activateOnApproval = this.fromApiJsonHelper.extractBooleanNamed(activateOnApprovalParameterName, element);
+
+        if (activateOnApproval == null || !activateOnApproval) {
+            validateRepaymentsStartingFromDateIsAfterDisbursementDate(dataValidationErrors, expectedDisbursementDate,
```

Comment:
Is there a reason why this validation is only used on CalculateLoanSchedule and not on LoanApplication?